### ISSUE TITLE
FP S2259: optional chaining

### ISSUE THREAD

**Issue by vilchik-elena** [SonarSource/SonarJS#2921]
```js
let x;

if (condition) {
  x?.foo(); // Should not be reported
}
```

Reported in https://community.sonarsource.com/t/fp-s2259-with-multiple-optional-chaining-in-javascript/54014
